### PR TITLE
Add `is_finite_dimensional_vector_space`

### LINKED ISSUE #3510: vector_space_dimension returning infinity instead of throwing an error
**Is your feature request related to a problem? Please describe.**
Working with vector_space_dimension can be annoing since it throws an error exception in the infinite dimensional case. 
You always have to treat the infinite dimensional case separately, even if it would not be nessesary like for example if you want to check if two MPolyQuoRings have the same vector space dimension.  When doing simple things like that you could easily forget to treat the infinite dimensional case at all.

**Describe the solution you'd like**
 vector_space_dimension(A::MPolyQuoRing)  should in my opinion return infinity instead of throwing an error exception if A is a infinite dimensional vectorspace.

**Describe alternatives you've considered**
A minimal solution would be to provide a function is_finite_dimensional_vector_space().

### PR DESCRIPTION
**Pull request by joschmitt**
Resolves #3510.

Implements what I believe to be the result of the discussion in #3510: A function `is_finite_dimensional_vector_space(::MPolyQuoRing)` is added and `vector_space_dimension` now throws an error of type `InfiniteDimensionError` if the dimension is infinite.

For completeness: `vector_space_dimension` also exists for certain localized rings and modules. In theory, we could/should add `is_finite_dimensional_vector_space` also for those, but it seems to me that in those cases `is_finite_dimensional_vector_space` would do some non-trivial computations which have to be repeated by `vector_space_dimension`.

### CODE REVIEW

**Comment by afkafkafk13** [comment]
> Resolves #3510.
> 
> Implements what I believe to be the result of the discussion in #3510: A function `is_finite_dimensional_vector_space(::MPolyQuoRing)` is added and `vector_space_dimension` now throws an error of type `InfiniteDimensionError` if the dimension is infinite.
> 
> For completeness: `vector_space_dimension` also exists for certain localized rings and modules. In theory, we could/should add `is_finite_dimensional_vector_space` also for those, but it seems to me that in those cases `is_finite_dimensional_vector_space` whould do some non-trivial computations which have to be repeated by `vector_space_dimension`.

Indeed. Your fix now raises an appropriate error also in the local case, as this is referred to the leading ideal of the  "shifted ideal" and the global case.

For 'is_finite_dimensional_vector_space' in the local setting, we would need to cache the leading ideal of the shifted ideal -- which makes a lot of sense anyway. But I would prefer to discuss this also with @HechtiDerLachs , when he is back from his holidays. He designed the localizations and know the pitfalls best, which might appear, if we take this route.

**Comment by afkafkafk13** [comment]
P.S.: Please add is_finite_dimensional_vector_space returning false for MPolyAnyNonQuoRing and one which says 'not implemented yet' for the localized quotient ring case.
Edit: This is to make it look agnostic for the enduser.

File: src/Rings/mpoly-affine-algebras.jl [hunk 1]
```diff
@@ -19,11 +19,46 @@ julia> dim(A)
...
+false
+```
+"""
+function is_finite_dimensional_vector_space(A::MPolyQuoRing)
+  # We check '<=' because A might be the zero ring, so dim(A) == -1
+  return dim(A) <= 0
+end
+
+struct InfiniteDimensionError <: Exception
+    InfiniteDimensionError() = new()
```
Comment (thofma) [inline src/Rings/mpoly-affine-algebras.jl]:
```suggestion
```
Should not be necessary.

**Comment by joschmitt** [comment]
> P.S.: Please add is_finite_dimensional_vector_space returning false for MPolyAnyNonQuoRing and one which says 'not implemented yet' for the localized quotient ring case. Edit: This is to make it look agnostic for the enduser.

Good idea, I added those functions.

**Comment by afkafkafk13** [comment]
Thank you. Looks good now. I will approve it, as soon as I have seen the output of the tests.
By the way: why are they stalled right now?

**Review by afkafkafk13 — APPROVED** [review]

**Review by fingolfin — APPROVED** [review]
Thank you

**Pull request merged**